src/chapters/more-on-functions/recursion-walkthrough.rst

```diff
@@ -0,0 +1,207 @@
+
+   * - .. sourcecode:: js
+
+         let arr = ['L', 'C', '1', '0', '1'];
+         let newString = '';
+
+         for (i = 0; i < arr.length; i++){
+            newString = newString + arr[i];
+         }
+
+         console.log(newString);
+         console.log(arr);
+
+       **Output**
+
+       .. sourcecode:: js
+
+         'LC101'
+         ['L', 'C', '1', '0', '1']
+
+     - .. sourcecode:: js
+
+         let arr = ['L', 'C', '1', '0', '1'];
+         let newString = '';
+
+         while (arr.length > 0){
+            newString += arr[0];
+            arr.shift();
+         }
+         console.log(newString);
+         console.log(arr);
+
+       **Output**
+
+       .. sourcecode:: js
+
+         'LC101'
+         [ ]
+
+Inside each loop, the code simply adds two strings together - whatever is
+stored in ``newString`` plus one element from the array. In the ``for`` loop,
+the element is the next item in the sequence of entries.  In the ``while``
+loop, the element is always the first entry from whatever remains in the array.
+
+OK, the loops join the array elements together. Now let's see how to
+accomplish the same task without a ``for`` or ``while`` statement.
+
+Bring In Recursion Concepts
+----------------------------
+
+First, state the problem to solve: *Combine the elements from an array into a
+string*.
+
+Second, split the problem into small, identical steps: Looking at the loops
+above, the "identical step" is just adding two strings together - ``newString``
+and the next entry in the array.
+
+Third, build a function to accomplish the small steps: Let's call the function
+``combineEntries``, and we will set an array as the parameter.
+
+.. sourcecode:: js
+
+   function combineEntries(arrayName){
+      //some code here
+   }
+
+We want ``combineEntries`` to repeat over and over again until the task is
+complete.
+
+How do we make this happen without using ``for`` or ``while``?
+
+.. index:: ! base case
+
+Identifying the Base Case
+--------------------------
+
+| ``for`` and ``while`` loops end when a particular condition evaluates to
+   ``false``. In the examples above, these conditions are
+| ``i < arr.length`` and ``arr.length > 0``, respectively.
+
+With recursion, we do not know how many times ``combineEntries`` must be
+called. To make sure the code stops at the proper time, we need to identify a
+condition that ends the process. This is called the **base case**, and it
+represents the simplest possible task for our function.
+
+``if`` the base case is ``true``, the recursion ends and the task is complete.
+``if`` the base case is ``false``, the function calls itself again.
+
+We check for the base case like this:
+
+.. sourcecode:: js
+
+   function combineEntries(arrayName){
+      if (baseCase is true){
+         //solve last step
+         //end recursion
+      } else {
+         //solve next small step
+         //call combineEntries again
```

Comment:
These two lines of comments feel off. Saying "solve the next step" implies that some resolution to the problem will be found, and _then_ the function calls itself. Rather than finding some solution, we are reducing the problem down into a smaller version of the same problem, and then the function calls itself.

Reply:
Fixed.  ``else`` now just calls ``combineEntries`` again.